src/MonoTorrent/MonoTorrent.Client/Modes/Mode.cs

```diff
@@ -532,9 +531,11 @@ void PreLogicTick (int counter)
             if (Manager.finishedPieces.Count > 0)
                 SendHaveMessagesToAll ();
 
-            for (int i = 0; i < Manager.Peers.ConnectedPeers.Count; i++) {
-                id = Manager.Peers.ConnectedPeers[i];
-                if (id.Connection == null)
+            foreach (PeerId id in Manager.Peers.ConnectedPeers.ToArray())
```

Comment:
I don't want to repeat too much, but the `ToArray` shouldn't be necessary anywhere and could result in a fairly high number of allocations :)